app-backend/api/src/main/scala/scene/Routes.scala

```diff
@@ -328,47 +314,44 @@ trait SceneRoutes extends Authentication
   // Safe to `bands.head`, since construction of a MultibandTile requires at least one band, so it's
   // impossible to have an empty one
   @SuppressWarnings(Array("TraversableHead"))
-  def getSceneThumbnail(sceneId: UUID): Route = authenticateWithParameter { user =>
-    {
-      thumbnailQueryParameters {
-        thumbnailParams => {
-          authorizeAsync {
-            SceneDao.authViewQuery(user, ObjectType.Scene)
-              .filter(sceneId)
-              .exists
-              .transact(xa)
-              .unsafeToFuture
-          } {
-            complete {
-              SceneDao.unsafeGetSceneById(sceneId).transact(xa).unsafeToFuture >>=
-                { (scene: Scene) =>
-                  (scene.ingestLocation, scene.sceneType) match {
-                    case (Some(uri), Some(SceneType.COG)) => {
-                      CogUtils.thumbnail(
-                        uri, thumbnailParams.width, thumbnailParams.height,
-                        thumbnailParams.red, thumbnailParams.green, thumbnailParams.blue,
-                        thumbnailParams.floor
-                      ).map(
-                        (tile: MultibandTile) => {
```

Comment:
This is the location of the other **Match instead of partial function** warning that I was unable to resolve.

Reply:
There's a good `cats` solution for this that I didn't know about at the time. I'll push up a commit with it if you don't mind.

Reply:
Yeah it does, check out https://github.com/raster-foundry/raster-foundry/pull/3891/commits/60f8cfbc81912c64cc11b791f800b4c27ec5e056#diff-05be688c0243d22c8c1381c201d2a365R330

Reply:
😥